src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -418,28 +448,157 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
     else:
         raise CLIError('DNS routing verification failed')
 
+# Validate --image input
+def validate_image(cmd,
+                   registry_name,
+                   repository_name,
+                   login_server,
+                   username,
+                   password,
+                   image):
+    from .manifest import _get_v2_manifest_path
+
+    from ._docker_utils import (
+        request_data_from_registry,
+        get_manifest_authorization_header,
+        RegistryException
+    )
+
+    try:
+        request_data_from_registry(
+            http_method='get',
+            login_server=login_server,
+            path=_get_v2_manifest_path(repository_name, image),
+            username=username,
+            password=password,
+            json_payload=None,
+            file_payload=None,
+            manifest_headers=get_manifest_authorization_header(username, password),
+            params=image)
+
+    except RegistryException as e:
+        if e.status_code == 404:
+            _check_image_match(cmd,
+                               registry_name,
+                               repository_name,
+                               login_server,
+                               username,
+                               password,
+                               image)
+        raise CLIError("{} Please check if image/tag was inputted correctly.".format(str(e)))
+
+# Validate if blob layer can be pulled
+def _check_blob(repository_name,
+                login_server,
+                username,
+                password,
+                image):
+
+    import requests
+    from ._docker_utils import get_manifest_authorization_header, parse_error_message
+
+    # Get manifest
+    # GET {url}/v2/{name}/manifests/{reference}
+    manifest_url = 'https://{}/v2/{}/manifests/{}'.format(login_server, repository_name, image)
+
+    manifest_response = requests.get(
+        manifest_url,
+        headers=get_manifest_authorization_header(username, password))
+
+    manifest = manifest_response.json()
+
+    # Get the digest of the smallest blob for performance purposes
+    smallest_blob = min(manifest['layers'], key=lambda layer: layer['size'])
```

Comment:
would this work any given manifest? Did you try with a Docker Manifest list?

Reply:
Good point, I would need to check if the manifest is a manifest list first by checking the media type layer, then find the digest of the smallest blob layer. Will rewrite and test this, thank you.